climada_petals/hazard/tc_rainfield.py

```diff
@@ -65,200 +214,1552 @@ def set_from_tracks(self, *args, **kwargs):
+        centroids: Optional[Centroids] = None,
+        pool: Optional[pathos.pools.ProcessPool] = None,
+        model: str = 'R-CLIPER',
+        model_kwargs: Optional[dict] = None,
+        ignore_distance_to_coast: bool = False,
+        store_rainrates: bool = False,
+        metric: str = "equirect",
+        intensity_thres: float = DEF_INTENSITY_THRES,
+        max_latitude: float = 61,
+        max_dist_inland_km: float = 1000,
+        max_dist_eye_km: float = DEF_MAX_DIST_EYE_KM,
+        max_memory_gb: float = DEF_MAX_MEMORY_GB,
+    ):
+        """
+        Create new TCRain instance that contains rainfields from the specified tracks
+
+        This function sets the `intensity` attribute to contain, for each centroid,
+        the total amount of rain experienced over the whole period of each TC event in mm.
+        The amount of rain is set to 0 if it does not exceed the threshold `intensity_thres`.
+
+        The `category` attribute is set to the value of the `category`-attribute
+        of each of the given track data sets.
+
+        The `basin` attribute is set to the genesis basin for each event, which
+        is the first value of the `basin`-variable in each of the given track data sets.
+
+        Optionally, the time-dependent rain rates can be stored using the `store_rainrates`
+        function parameter (see below).
+
+        Currently, two models are supported to compute the rain rates: R-CLIPER and TCR. The
+        R-CLIPER model is documented in Tuleya et al. 2007. The TCR model was used by
+        Zhu et al. 2013 and Emanuel 2017 for the first time and is documented in detail in
+        Lu et al. 2018. This implementation of TCR includes improvements proposed in
+        Feldmann et al. 2019. TCR's accuracy is much higher than R-CLIPER's at the cost of
+        additional computational and data requirements.
+
+        When using the TCR model make sure that your TC track data includes the along-track
+        variables "t600" (temperature at 600 hPa) and "u850"/"v850" (wind speed at 850 hPa). Both
+        can be extracted from reanalysis or climate model outputs. For "t600", use the value at the
+        storm center. For "u850"/"v850", use the average over the 200-500 km annulus around the
+        storm center. If "u850"/"v850" is missing, this implementation sets the shear component of
+        the vertical velocity to 0. If "t600" is missing, the saturation specific humidity is set
+        to a universal estimate of 0.01 kg/kg. Both assumptions can have a large effect on the
+        results (see Lu et al. 2018).
+
+        The implementation of the R-CLIPER model currently does not allow modifications, so that
+        `model_kwargs` is ignored with `model="R-CLIPER"`. While the TCR model can be configured in
+        several ways, it is usually safe to go with the default settings. Here is the complete list
+        of `model_kwargs` and their meaning with `model="TCR"` (in alphabetical order):
```

Comment:
I'd expect the explanation of what _model_kwargs_ is about in the function's parameters description just below, under _model_kwargs_.

Reply:
Good point, I moved the description accordingly.